package proto_ff;

// calm 自定义的fieldoption都在该文件中
import "yd_fieldoptions.proto";
import "common_logic.proto";

import "Com.proto";

//返回 玩家信息 CLIENT_PLAYER_INFO_RSP
message	PlayerInfoRsp
{
  optional uint64 cid = 1;						//cid
  optional bytes name = 2;						//名字
  optional uint32 zid = 3;						//区服ID
  optional uint64 opentime = 4;					//区服开服 UTC 时间，
  optional uint64 createtime = 5;					//角色创建时间
  optional RoleFacadeProto facade = 6;			//角色外观
  repeated Attr64 attr = 7;						//属性
  optional uint64 mapid = 8;						//地图ID
  optional Vector3PB pos = 9;						//坐标
  optional FunctionUnlockInfo unlockInfo	= 10;	//功能解锁信息
  optional int32 state = 11;						//状态，详见 Com.proto中的枚举 ECState
  optional SkillGroupListProto skill_group = 12;	//技能组信息
}



//生物属性同步 CREATURE_ATTR_SYN
message CreatureAttrSyn
{
  optional uint64 cid = 1;			//cid
  repeated Attr64 attr = 2;			//属性列表
}

//生物属性广播 CREATURE_ATTR_BROAD
message CreatureAttrBroadRsp
{
  optional uint64 cid = 1;			//cid
  repeated Attr64 attr = 2;			//属性列表
}

//生物状态广播 CREATURE_STATE_BROAD
message CreatureStateBroadRsp
{
  optional uint64 cid = 1;			//cid
  optional uint32 curstate = 2;		//当前状态
  optional uint32 beforestate = 3;	//前一个状态
}

//通知客户端显示信息(主要用于客户端提示、飘字等显示用)  CLIENT_SHOW_INFO_RSP
message NoticeShowInfoRsp
{
  optional int32 show_type = 1;				 //显示类型 1-经验, 2-掉落物品(前端掉落飘入背包的效果)
  optional uint64 add_val = 2;				 //增加的值
  optional MultItemSimpleProto item_lst = 3;   //物品ID列表(显示类型为2的时候用到)
}

//请求 角色改名 CLIENT_CHG_NAME_REQ
message ChgNameReq
{
  optional string name = 1;        //新名字
}
//返回 角色改名 结果 CLIENT_CHG_NAME_RSP
message ChgNameRsp
{
  required int32 retcode = 1;  //返回码
  optional bytes name = 2;		 //新名字
}


//通知客户端死亡复活 PLAYER_RELIVE_NOTIFY
message PlayerReliveNotify
{
  optional bytes name = 1;		//被谁击杀
  optional int32 left_num = 2;    //再经过多少次复活就会进入复活疲劳
  optional int32 sec = 3;			//复活倒计时,单位：秒
  optional bool is_tired = 4;		//是否处于疲劳状态
  optional int32 already_num = 5;	//当前地图中已经复活的次数
}

//请求复活  PLAYER_RELIVE_REQ
message PlayerReliveReq
{
  optional int32 type = 1; //复活类型 1-普通复活，2-原地复活
}
//返回 复活结果  PLAYER_RELIVE_RSP
message PlayerReliveRsp
{
  required int32 ret = 1; //返回码
}